clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -4853,9 +4853,16 @@ void UnwrappedLineParser::readToken(int LevelDifference) {
     PreviousWasComment = FormatTok->is(tok::comment);
 
     while (!Line->InPPDirective && FormatTok->is(tok::hash) &&
-           (!Style.isVerilog() ||
-            Keywords.isVerilogPPDirective(*Tokens->peekNextToken())) &&
            FirstNonCommentOnLine) {
+      // In Verilog, the backtick is used for macro invocations. In TableGen,
+      // the single hash is used for the paste operator.
+      const FormatToken *Next = Tokens->peekNextToken();
+      assert(Next); // There is an EOF token at the end.
```

Comment:
```suggestion
      const auto *Next = Tokens->peekNextToken();
```
It seems that `peekNextToken()` never returns null.

Reply:
I thought that asserts were for things that never happen.  So I added one.  Is this something that should happen even less so that asserts are no good?

Reply:
IMO `assert()` should *not* be used on something that's obviously true, and for this reason we never assert on `peekNextToken()`.